modules/service/src/main/scala/lucuma/itc/service/ItcMappings.scala

```diff
@@ -64,22 +67,22 @@ import QueryCompiler._
 trait Encoders {
   import io.circe.generic.semiauto._
   import io.circe.syntax._
-  type Nanosecond  = Nano %* Second
-  type Microsecond = Micro %* Second
-  type Millisecond = Milli %* Second
+  type Nanosecond  = Nano * Second
+  type Microsecond = Micro * Second
+  type Millisecond = Milli * Second
 
   implicit val encoderFiniteDuration: Encoder[FiniteDuration] = new Encoder[FiniteDuration] {
     type Micro
-    final def apply(d: FiniteDuration): Json = {
-      val value = d.toNanos.withUnit[Nanosecond]
+    final def apply(d: FiniteDuration): Json =
+      val value: Quantity[Long, Nanosecond] = d.toNanos.withUnit[Nanosecond]
+
       Json.obj(
-        ("microseconds", Json.fromLong(value.toUnit[Microsecond].value)),
-        ("milliseconds", Json.fromBigDecimal(value.to[BigDecimal, Millisecond].value)),
-        ("seconds", Json.fromBigDecimal(value.to[BigDecimal, Second].value)),
-        ("minutes", Json.fromBigDecimal(value.to[BigDecimal, Minute].value)),
-        ("hours", Json.fromBigDecimal(value.to[BigDecimal, Hour].value))
+        ("microseconds", Json.fromLong(value.toValue[BigDecimal].toUnit[Microsecond].value.toLong)),
```

Comment:
I think you can use `.tToUnit` to indicate truncated conversion.

```suggestion
        ("microseconds", Json.fromLong(value.tToUnit[Microsecond].value)),
```